### PR TITLE
refactor(menulist): replace sass with styled component

### PR DESCRIPTION
**Pull request by sjohnson-sage**
### Proposed behaviour
Replace SCSS with styled component file and add a deprecation warning.

image

Also replaced `package.json`s with `index.js`s and some other renaming files to tidy up
### Current behaviour
Component uses sass for its styling.

### Checklist

- [x] Commits follow our style guide
- [x] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [x] Typescript `d.ts` file added or updated if required
- [x] Carbon implementation and Design System documentation are congruent

### Testing instructions
http://localhost:9001/?path=/docs/menulist--default-story

### CODE REVIEW

File: src/components/menu-list/menu-list.stories.mdx [hunk 1]
```diff
@@ -1,179 +1,173 @@
...
+    style={{
+      backgroundColor: "red",
+      textAlign: "center",
+      color: "white",
+      padding: 20,
+      fontWeight: "bold",
+      marginBottom: 10,
+    }}
+  >
+    Create has been deprecated please see
```
Comment (nicktitchmarsh) [inline src/components/menu-list/menu-list.stories.mdx:24]: Should probably add correct wording in here as we aren't converting a button

Comment (nicktitchmarsh) [inline src/components/menu-list/menu-list.stories.mdx:24]: This should reference MenuList not Create

Comment (sjohnson-sage) [reply src/components/menu-list/menu-list.stories.mdx:24]: Ah, forgot to update and copying and pasting in!

**Review by nicktitchmarsh — APPROVED** [review]

File: src/components/menu-list/menu-list.stories.mdx [hunk 2]
```diff
@@ -1,179 +1,164 @@
-import { useState } from 'react';
-import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
-import { MenuList, MenuListItem } from '.'
+import { useState } from "react";
+import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
+import { MenuList, MenuListItem } from ".";
 
-<Meta title="MenuList" parameters={{ info: { disable: true }}} />
+<Meta title="MenuList" parameters={{ info: { disable: true } }} />
```
Comment (edleeks87) [inline src/components/menu-list/menu-list.stories.mdx:5]: Should we disable the chromatic across all these stories now it's being deprecated?

**Review by nicktitchmarsh — DISMISSED** [review]

**Review by edleeks87 — DISMISSED** [review]

**Review by Alec-Walker-Sage — DISMISSED** [review]

File: cypress/features/regression/menuList.feature [hunk 3]
```diff
@@ -10,11 +10,11 @@ Feature: MenuList component
       | mp150ú¿¡üßä                  | titleOtherLanguage    |
       | !@#$%^*()_+-=~[];:.,?{}&"'<> | titleSpecialCharacter |
 
-  @ignore
-  # @positive doesn't work on Carbon Demo site
-  Scenario Outline: Change MenuList filterPlaceholder to <text>
+
```
Comment (Alec-Walker-Sage) [inline cypress/features/regression/menuList.feature:13]: Spacing

**Review by KatarzynaQA — DISMISSED** [review]

**Review by nicktitchmarsh — APPROVED** [review]

**Review by edleeks87 — APPROVED** [review]

**Review by Alec-Walker-Sage — APPROVED** [review]

**Pull request merged**